src/OSPSuite.Core/Domain/ModelCoreSimulationExtensions.cs

```diff
@@ -30,9 +30,18 @@ public static class ModelCoreSimulationExtensions
          return AllApplicationParametersOrderedByStartTimeFor(simulation, moleculeName);
       }
 
+      private static List<EventGroup> getEventGroupsFrom(IContainer container)
+      {
+         return container?.GetChildren<EventGroup>().ToList();
+      }
+
       private static IReadOnlyList<IContainer> allApplicationsForMolecule(IModelCoreSimulation simulation, string moleculeName)
       {
-         var applicationEventGroup = simulation.Model.Root.GetChildren<EventGroup>().ToList();
+         var applicationsFromEventContainer = getEventGroupsFrom(simulation.Model.Root.GetSingleChildByName<IContainer>(Constants.EVENTS));
```

Comment:
So, yeah, for now, I just combined the two known locations where we might find applications.

Reply:
Try to remove the knowledge of container name by getting all children of type eventgroup. Check that this method is called sparingly.

Reply:
Or maybe the 'Events' container should be of type 'EventGroup' instead of other